README.md

```diff
@@ -737,3 +739,103 @@ pipe(
   Option.option.traverse(TaskEither.taskEither) // this is the same as above.
 )
 ```
+
+## <a name="reader"></a>Reader
+
+If you already had to carry dependencies throught out multiple functions, you will quickly understand the value of `Reader`.
+You can see the definition given by Giulio Canti [here](https://dev.to/gcanti/getting-started-with-fp-ts-reader-1ie5), and basically `Reader<R,A>` represents a function `(r: R) => A` where `R` will be your dependencies, and A the result.
+
+Let's take an example:
+
+```typescript
+const foo = ({
+  firstId,
+  secondId,
+  eventData,
+  firstDependency,
+  secondDependency,
+}): TaskEither<Error, Value> =>
+  pipe(
+    firstDependency.getById(firstId),
+    TaskEither.chain(() => secondDependency.sendEvent({ secondId, eventData }))
+  );
+```
+
+Using `Reader`, you could write it this way:
+
+```typescript
+const foo = ({
+  firstId,
+  secondId,
+  eventData,
+}): ReaderTaskEither<Deps, Error, Value> =>
+  pipe(
+    getById(firstId),
+    ReaderTaskEither.chain(() => sendEvent({ secondId, eventData }))
+  );
+
+// with:
+const getById = (firstId) =>
+  pipe(
+    Reader.ask<Deps>(),
+    Reader.map(({ firstDependency }) => firstDependency.getById(firstId))
+  );
+const sendEvent = ({ secondId, eventData }) =>
+  pipe(
+    Reader.ask<Deps>(),
+    Reader.map(({ secondDependency }) =>
+      secondDependency.sendEvent({ secondId, eventData })
+    )
+  );
+```
+
+You need to adapt some methods to be able to properly use the `Reader` pattern, but once it's done, you won't need to carry your dependencies from a method to another as those will only be seen where they are needed.
+
+## <a name="specific-examples"></a> Some specific examples
+
+### <a name="call-in-parallel"></a> Call in parallel
+
+At some point, you might need to call methods in parallel, for instance because you need to send data to multiple external services and you don't want to decrease the performance of your use case. Here is an example that shows you how to call two `ReaderTaskEither` in parallel :
+
+```typescript
+import * as RTE from "fp-ts/ReaderTaskEither";
+import { pipe } from "fp-ts/function";
+
+declare const foo: RTE.ReaderTaskEither<R1, E1, A1>;
+declare const bar: RTE.ReaderTaskEither<R2, E2, A2>;
+
+pipe(
+  RTE.Do,
+  RTE.bindW("foo", () => foo),
```

Comment:
```suggestion
  RTE.apS("foo", foo),
```
It does not change much, as far as I know, but I find it clearer like this.
anyway, you can get rid of the `W` if you keep the `bind`